src/Uno.UI/UI/Xaml/Controls/MenuFlyout/MenuFlyoutItem.cs

```diff
@@ -73,5 +139,685 @@ internal void InvokeClick()
+			 bool bUseTransitions)
+		{
+			bool hasToggleMenuItem = false;
+			bool hasIconMenuItem = false;
+			bool hasMenuItemWithKeyboardAcceleratorText = false;
+			bool isKeyboardPresent = false;
+
+			MenuFlyoutPresenter spPresenter = GetParentMenuFlyoutPresenter();
+			if (spPresenter != null)
+			{
+				hasToggleMenuItem = spPresenter.GetContainsToggleItems();
+				hasIconMenuItem = spPresenter.GetContainsIconItems();
+				hasMenuItemWithKeyboardAcceleratorText = spPresenter.GetContainsItemsWithKeyboardAcceleratorText();
+			}
+
+			var bIsEnabled = IsEnabled;
+			var focusState = FocusState;
+			var shouldBeNarrow = GetShouldBeNarrow();
+
+			// We only care about finding if we have a keyboard if we also have a menu item with accelerator text,
+			// since if we don't have any menu items with accelerator text, we won't be showing any accelerator text anyway.
+			if (hasMenuItemWithKeyboardAcceleratorText)
+			{
+				// UNO TODO
+				// isKeyboardPresent = DXamlCore.GetCurrent().GetIsKeyboardPresent();
+				isKeyboardPresent = true;
+			}
+
+			// CommonStates
+			if (!bIsEnabled)
+			{
+				VisualStateManager.GoToState(this, "Disabled", bUseTransitions);
+			}
+			else if (m_bIsPressed)
+			{
+				VisualStateManager.GoToState(this, "Pressed", bUseTransitions);
+			}
+			else if (m_bIsPointerOver)
+			{
+				VisualStateManager.GoToState(this, "PointerOver", bUseTransitions);
+			}
+			else
+			{
+				VisualStateManager.GoToState(this, "Normal", bUseTransitions);
+			}
+
+			// FocusStates
+			if (FocusState.Unfocused != focusState && bIsEnabled)
+			{
+				if (FocusState.Pointer == focusState)
+				{
+					VisualStateManager.GoToState(this, "PointerFocused", bUseTransitions);
+				}
+				else
+				{
+					VisualStateManager.GoToState(this, "Focused", bUseTransitions);
+				}
+			}
+			else
+			{
+				VisualStateManager.GoToState(this, "Unfocused", bUseTransitions);
+			}
+
+			// CheckPlaceholderStates
+			if (hasToggleMenuItem && hasIconMenuItem)
+			{
+				VisualStateManager.GoToState(this, "CheckAndIconPlaceholder", bUseTransitions);
+			}
+			else if (hasToggleMenuItem)
+			{
+				VisualStateManager.GoToState(this, "CheckPlaceholder", bUseTransitions);
+			}
+			else if (hasIconMenuItem)
+			{
+				VisualStateManager.GoToState(this, "IconPlaceholder", bUseTransitions);
+			}
+			else
+			{
+				VisualStateManager.GoToState(this, "NoPlaceholder", bUseTransitions);
+			}
+
+			// PaddingSizeStates
+			if (shouldBeNarrow)
+			{
+				VisualStateManager.GoToState(this, "NarrowPadding", bUseTransitions);
+			}
+			else
+			{
+				VisualStateManager.GoToState(this, "DefaultPadding", bUseTransitions);
+			}
+
+			// We'll make the accelerator text visible if any item has accelerator text,
+			// as this causes the margin to be applied which reserves space, ensuring that accelerator text
+			// in one item won't be at the same horizontal position as label text in another item.
+			if (hasMenuItemWithKeyboardAcceleratorText && isKeyboardPresent)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Change this condition so that it does not always evaluate to 'true'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5205492)